hypothesis-python/src/hypothesis/internal/conjecture/utils.py

```diff
@@ -34,6 +36,20 @@ def calc_label_from_name(name: str) -> int:
     return int_from_bytes(hashed[:8])
 
 
+def calc_label_from_callable(f: Callable) -> int:
```

Comment:
I was somewhat worried about the performance implications of this for `st.composite`, which creates a new strategy every time it's called. It seems fine though, I don't think a cache is necessary yet.

Reply:
nice, that's certainly in the acceptable range